File: leetcode/Easy/0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
  vector<vector<int>> generate(int numRows) {
    vector<vector<int>> answer;
    if (numRows == 0)
      return answer;
    if (numRows == 1) {
      answer.push_back({1});
      return answer;
    }
    answer = generate(numRows - 1);
    vector<int> line(1, 1);
    for (int i = 1; i < numRows - 1; i++) {
      line.push_back(answer[numRows - 2][i] + answer[numRows - 2][i - 1]);
    }
    line.push_back(1);
    answer.push_back(line);
    return answer;
  }
};

File: leetcode/Easy/0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        if numRows == 1:
            return [[1]]
        ans = self.generate(numRows - 1)
        line = [1]
        for i in range(1, numRows - 1):
            line.append(ans[numRows - 2][i] + ans[numRows - 2][i - 1])
        line.append(1)
        ans.append(line)
        return ans
